src/coreclr/tools/Common/TypeSystem/Common/TypeHashingAlgorithms.cs

```diff
@@ -36,35 +37,29 @@ public void Append(string src)
                 int startIndex = 0;
                 if ((_numCharactersHashed & 1) == 1)
                 {
-                    _hash2 = (_hash2 + _rotl(_hash2, 5)) ^ src[0];
+                    _hash2 = (_hash2 + (int)BitOperations.RotateLeft((uint)_hash2, 5)) ^ src[0];
```

Comment:
Basically any of the `BitOperations` APIs where you're having to insert casts can be avoided by just calling the same API directly on the type. Not going to comment on all of them